modules/bluefish/util/blue_velvet.cpp

```diff
@@ -20,67 +20,231 @@
 */
 
 #include "../StdAfx.h"
-
 #include "blue_velvet.h"
+#include <common/utf.h>	
+#include <core/video_format.h>
+#include <bluefish/interop/BlueVelvetCUtils.h>
 
-#include <common/utf.h>
+#if defined(__APPLE__)
+	#include <dlfcn.h>
+#endif
 
-#include <core/video_format.h>
+#if defined(_WIN32)
+	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)GetProcAddress(module, #name); if(!name) { return false; } }
+#elif defined(__APPLE__)
+	#define GET_PROCADDR_FOR_FUNC(name, module) { name = (pFunc_##name)dlsym(module, #name); if(!name) { return false; } }
+#endif
 
 namespace caspar { namespace bluefish {
-	
-CBlueVelvet4* (*BlueVelvetFactory4)() = nullptr;
-void (*BlueVelvetDestroy)(CBlueVelvet4* pObj) = nullptr;
-const char*	(*BlueVelvetVersion)() = nullptr;
-BLUE_UINT32 (*encode_hanc_frame)(struct hanc_stream_info_struct * hanc_stream_ptr, void * audio_pcm_ptr,BLUE_UINT32 no_audio_ch,BLUE_UINT32 no_audio_samples,BLUE_UINT32 nTypeOfSample,BLUE_UINT32 emb_audio_flag) = nullptr;
-BLUE_UINT32 (*encode_hanc_frame_ex)(BLUE_UINT32 card_type, struct hanc_stream_info_struct * hanc_stream_ptr, void * audio_pcm_ptr, BLUE_UINT32 no_audio_ch,	BLUE_UINT32 no_audio_samples, BLUE_UINT32 nTypeOfSample, BLUE_UINT32 emb_audio_flag) = nullptr;
 
-void blue_velvet_initialize()
-{
-#ifdef _DEBUG
-	std::string module_str = "BlueVelvet64_d.dll";
-#else
-	std::string module_str = "BlueVelvet64.dll";
+	BvcWrapper::BvcWrapper()
+	{
+		bvc = nullptr;
+		hModule = nullptr;
+		InitFunctionsPointers();
+		if (bfcFactory)
```

Comment:
Currently this is undefined behaviour since `bfcFactory` is not initialized to `nullptr`. This should not be needed anyway if my other suggestions about throwing an exception during `InitFunctionsPointers()`